src/group/mod.rs

```diff
@@ -1,71 +1,175 @@
+use openmls::{
+    prelude::Verifiable,
+    prelude::{
+        group_info::{GroupInfo, VerifiableGroupInfo},
+        CreationFromExternalError, ProposalStore, PublicGroup,
+    },
+    prelude::{
+        ConfirmationTag, LibraryError, OpenMlsSignaturePublicKey, ProcessedMessage,
+        ProcessedMessageContent, Sender, StagedCommit,
+    },
+    treesync::{LeafNode, Node},
+};
+use openmls_rust_crypto::{OpenMlsCryptoProvider, OpenMlsRustCrypto};
 use serde::{Deserialize, Serialize};
-use tls_codec::{TlsDeserialize, TlsSerialize, TlsSize, VLBytes};
 
-use crate::{
-    messages::AssistedMessage,
-    pool::{Ciphersuite, Extension, ProtocolVersion},
-};
+use crate::messages::{AssistedCommit, AssistedGroupInfo, AssistedMessage};
 
-pub mod process;
-mod validate_application;
-mod validate_commit;
-mod validate_proposal;
+use self::errors::ProcessAssistedMessageError;
 
+pub mod errors;
+
+#[derive(Serialize, Deserialize)]
 pub struct Group {
+    public_group: PublicGroup,
     group_info: GroupInfo,
+    #[serde(skip)]
+    backend: OpenMlsRustCrypto,
 }
 
 impl Group {
-    pub fn merge_staged_commit(&mut self, _staged_commit: StagedCommit) {}
-
-    pub fn public_tree() {}
+    /// Create a new group state with the group consisting of the creator's
+    /// leaf.
+    pub fn new(
+        verifiable_group_info: VerifiableGroupInfo,
+        leaf_node: LeafNode,
```

Comment:
In theory, this could also be a `KeyPackage`, no? But whatever works for now.

Reply:
Sure, we will have to change the API anyway when we start using mls-assist properly.